### PR TITLE
feat: automatic attestation for person credential

### PR DESCRIPTION
**Pull request by jleach**
When retrieving a person credentnial app attestation is required. This PR adds mechanics to automatically fetch the attestation credential if it is not present and accept the proof request.

### CODE REVIEW

File: app/src/helpers/Attestation.ts [hunk 1]
```diff
@@ -36,25 +41,157 @@ export interface AttestationProofRequestFormat {
...
+  // TODO:(jl) Should we be checking the length of the attributes matches some
+  // expected length in the proof request?
+
+  if (credentials.proofFormats.indy) {
+    return credentials.proofFormats.indy.attributes.attestationInfo.length == 0
+  }
+
+  if (credentials.proofFormats.anoncreds) {
+    return credentials.proofFormats.anoncreds.attributes.attestationInfo.length == 0
+  }
```
Comment (bryce-mcmath) [inline app/src/helpers/Attestation.ts]: should this be `!== <the number of attributes on our attestation credentials>`? Should be strict too I think (`!==` not `!=`)

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]
One initial change requested that we talked about yesterday. Other than that, does this logic handle both situations with the BCSC app? ie. does the person flow work both with and without attestation when attestation support is enabled in the wallet?

**Comment by jleach** [comment]
> One initial change requested that we talked about yesterday. Other than that, does this logic handle both situations with the BCSC app? ie. does the person flow work both with and without attestation when attestation support is enabled in the wallet?

The reason for the additional logic is that Credo does not select the newest non-revoked credential which we would like in all situations. I created [this](https://github.com/orgs/bcgov/projects/108?pane=issue&itemId=57285368) to track revisiting the issue to make sure it still persists in Credo 0.4.x and 0.5.x.

The Person workflow does work in both cases although once IDIM has deployed the change they will not issue a Person credential without a correct response to the proof.

File: app/src/helpers/Attestation.ts (hunk 1 above)
Comment (jleach) [reply app/src/helpers/Attestation.ts]: Will fix.

Comment (jleach) [reply app/src/helpers/Attestation.ts]: @bryce-mcmath I fixed the strict equality check. I think it should still be `===`. That credential will have no attributes if its missing. As part of a more advanced "walk" feature we should experiment to see if the proof is asking for 5 attributes then our fetched credential should have at least 5 attributes. I'm not sure if it works this way which is why I didn't add it now (probably needs a spike.

**Review by bryce-mcmath — APPROVED** [review]

**Pull request merged**